core/payments/ondemand/ondemandvalidation/on_demand_payment_validator.go

```diff
@@ -76,3 +76,23 @@ func (pv *OnDemandPaymentValidator) Debit(
 	pv.metrics.IncrementUnexpectedErrors()
 	return err
 }
+
+// RevertDebit reverts a previous debit operation for an on-demand payment
+// This is used when a blob dispersal fails after payment authorization
+func (pv *OnDemandPaymentValidator) RevertDebit(
```

Comment:
this "OnDemandPaymentValidator" is run on the disperser only right? and not on the validator?
I'm probably just dumb but "Validator" makes me think this is run on the validator itself. Maybe there's a way to remove this confusion, or maybe the confusion is only in my brain?